plugins/techdocs-backend/src/service/router.ts

```diff
@@ -127,6 +132,15 @@ export async function createRouter({
 
     const entity: Entity = await catalogRes.json();
 
+    if (!entity.metadata.uid) {
+      res.status(404).json({ error: 'Entity metadata UID missing' });
+      return;
+    }
+    if (!shouldCheckForUpdate(entity.metadata.uid)) {
+      res.status(200).json({ message: `Last update is recent, not retried` });
```

Comment:
```suggestion
      res.status(200).json({ message: `Last check for documentation update is recent, did not retry.` });
```

Reply:
What about https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/304

Reply:
I was wondering if 304 will cause specific browser behavior that we might not want. If that's not the case, yes 304 seems a better descriptor